src/server/api/user/validators.ts

```diff
@@ -5,14 +5,62 @@ import {
   isHttps,
   isPrintableAscii,
   isValidShortUrl,
+  isValidTag,
 } from '../../../shared/util/validation'
 import { LINK_DESCRIPTION_MAX_LENGTH } from '../../../shared/constants'
 import { isValidGovEmail } from '../../util/email'
 
+const MAX_TAG_COUNT = 3
+
 export const urlRetrievalSchema = Joi.object({
   userId: Joi.number().required(),
 })
 
+export const tagRetrievalSchema = Joi.object({
+  userId: Joi.number().required(),
+})
+
+export const userUrlsQueryConditions = Joi.object({
+  userId: Joi.number().required(),
+  limit: Joi.number().required(),
+  offset: Joi.number().optional(),
+  orderBy: Joi.string().valid('updatedAt', 'createdAt', 'clicks').optional(),
+  sortDirection: Joi.string().valid('desc', 'asc').optional(),
+  searchText: Joi.string().allow('').optional(),
+  state: Joi.string().allow('').optional(),
+  isFile: Joi.boolean().optional(),
+  tags: Joi.array()
+    .max(MAX_TAG_COUNT)
+    .optional()
+    .items(
+      Joi.string()
+        .optional()
+        .min(1)
+        .custom((tag: string, helpers) => {
+          if (!isValidTag(tag)) {
+            return helpers.message({ custom: `tag: ${tag} format is invalid` })
+          }
+          return tag
+        }),
+    ),
+})
```

Comment:
Thinking that we can create a object schema for tags and reuse it, see https://stackoverflow.com/a/42575904

Reply:
good idea, will look into this.

Reply:
done